wdl/transforms/draft3/src/main/scala/wdl/draft3/transforms/ast2wdlom/AstToWorkflowDefinitionElement.scala

```diff
@@ -0,0 +1,42 @@
+package wdl.draft3.transforms.ast2wdlom
+
+import cats.syntax.apply._
+import cats.syntax.either._
+import cats.syntax.validated._
+import common.validation.ErrorOr._
+import common.collections.EnhancedCollections._
+import wdl.draft3.parser.WdlParser.Ast
+import wdl.draft3.transforms.ast2wdlom.EnhancedDraft3Ast._
+import wdl.model.draft3.elements.{InputsSectionElement, OutputsSectionElement, WorkflowBodyElement, WorkflowDefinitionElement}
+
+
+object AstToWorkflowDefinitionElement {
+
+  def convert(a: Ast): ErrorOr[WorkflowDefinitionElement] = {
+
+    val nameElementValidation: ErrorOr[String] = astNodeToString(a.getAttribute("name")).toValidated
+
+    val bodyElementsValidation: ErrorOr[Vector[WorkflowBodyElement]] = a.getAttributeAsVector[WorkflowBodyElement]("body")(astNodeToWorkflowBodyElement).toValidated
+
+    (nameElementValidation, bodyElementsValidation) flatMapN { (name, bodyElements) =>
+      val inputs: Vector[InputsSectionElement] = bodyElements.filterByType[InputsSectionElement]
+      val outputs: Vector[OutputsSectionElement] = bodyElements.filterByType[OutputsSectionElement]
+
+      val inputsSectionValidation: ErrorOr[Option[InputsSectionElement]] = validateSize(inputs, "inputs", 1)
+      val outputsSectionValidation: ErrorOr[Option[OutputsSectionElement]] = validateSize(outputs, "outputs", 1)
+
+      (inputsSectionValidation, outputsSectionValidation) mapN { (validInputs, validOutputs) =>
+        WorkflowDefinitionElement(name, validInputs, validOutputs)
+      }
+    }
```

Comment:
ToL:

It might make things more readable to move everything from lines 22 to 31 into a separate function, maybe 
```
def combineElements(name: String, bodyElements: Vector[WorkflowBodyElement]): ErrorOr[WorkflowDefinitionElement] = { ... }
```
Then line 21 would become: 
```
(nameElementValidation, bodyElementsValidation) flatMapN combineElements
```